users-flask/app.py

```diff
@@ -0,0 +1,162 @@
+def handle_bad_request(e):
+    """Handles exception thrown when flask app is aborted using `abort` with HTTP status code 400."""
+    app.logger.exception(e)
+    data = CustomException(str(e)).json()
+    res = ResponseBodyJSON(False, data).json()
+    return jsonify(res), 400
+
+@app.errorhandler(KeyError)
+def handle_key_error(e: KeyError):
+    """Handles exception thrown when KeyError is thrown most probably when a field is missing in request body."""
+    app.logger.exception(e)
+    data = CustomException(f"Fields missing from request body {e.args}").json()
+    res = ResponseBodyJSON(False, data).json()
+    return jsonify(res), 400
+
+@app.errorhandler(IntegrityError)
+def handle_integrity_error(e: IntegrityError):
+    """Handles exception when integrity constraint in database are violated. E.g. not nullable fields are null (due to missing fields in request body)"""
+    app.logger.exception(e)
+    data = CustomException(f"Integrity constraint in database are violated. Did you check if all fields are present in request body? Error message: {str(e)}").json()
+    res = ResponseBodyJSON(False, data).json()
+    
+    return jsonify(res), 400
+
+@app.errorhandler(Exception)
+def handle_unhandled_exception(e: Exception):
+    app.logger.exception(e)
+    data = CustomException(str(e)).json()
+    res = ResponseBodyJSON(False, data).json()
+    return jsonify(res), 500
+
+
+@app.route('/users')
+def find_all_users():
+    args = request.args
+
+    milestone_id = args.get('milestone_id')
+    if milestone_id is not None:
+        offsets = db.session.execute(db.select(Offset).where(Offset.milestone_id == milestone_id)).scalars().all()
+        data = [offset.buyer.json() for offset in offsets]
+        res =  ResponseBodyJSON(True, data).json()
+        return jsonify(res), 200
+
+    else:
+        users = db.session.execute(db.select(User)).scalars().all()
+        data = [user.json() for user in users]
+        res =  ResponseBodyJSON(True, data).json()
+        return jsonify(res), 200
+    
+@app.route("/users/<uuid:id>")
+def find_user_by_id(id):
+    user = db.session.execute(db.select(User).where(User.id == id)).scalars().first()
+
+    if user is None:
+        abort(404, description=f"User {str(id)} not found.")
+    
+    data = user.json()
+    res = ResponseBodyJSON(True, data).json()
+    return jsonify(res), 200
+
+@app.route("/users", methods=['POST'])
+def create_user():
+    body = request.get_json()
+
+    user = User(**body)
+    db.session.add(user)
+    db.session.commit()
+    
+    data = user.json()
+    res = ResponseBodyJSON(True, data).json()
+    return jsonify(res), 201
+    
+
+@app.route("/users/verifiers")
+def find_verifiers():
+    verifiers = db.session.execute(db.select(User).where(User.is_verifier)).scalars().all()
+    data = [v.json() for v in verifiers]
+    res = ResponseBodyJSON(True, data).json()
+    return jsonify(res), 200
+
+
+@app.route("/users/<uuid:id>/offset", methods=['POST'])
```

Comment:
Postman test data for POST: offset - create one should be updated to the following
`
{
    "amount": 10.2,
    "buyer_id": "da686e54-83d2-471b-90f6-404e2077ac59",
    "created_at": "2023-03-31T18:48:39Z",
    "milestone_id": "ed9e5902-351d-4c72-9129-f98f5dcc1f7d",
    "payment_id": "pi_3MrCv6GehkGRMk671H2IKsDX",
    "status": "pending",
    "updated_at": "2023-03-31T18:48:39Z"
}
`